### ISSUE TITLE
Add analytics to AMP Stories created by the AMP plugin

### ISSUE THREAD

**Issue by westonruter** [google/site-kit-wp#435]
## Feature Description

When printing out analytics, currently the `amp-analytics` is output at the `wp_footer` action via:

https://github.com/google/site-kit-wp/blob/14d59a5fd00752f47ac955a5b65aacaa31d91b04/includes/Modules/TagManager.php#L186-L209

However, this will not end up adding analytics to AMP Stories, which do not trigger the `wp_footer` action triggered in its template:

https://github.com/ampproject/amp-wp/blob/575ad43026db5b44d2b17708b58a9f58f045b0b6/includes/templates/single-amp_story.php#L68-L74

We may introduce a `amp_story_footer` action, but this won't help much either because AMP Stories requires the `amp-analytics` to be placed _inside_ the `amp-story` element, not right before the `</body>` (as normal for `wp_footer`). Nevertheless, the `single-amp_story.php` template _does_ currently call `amp_print_analytics( '' );` right before `</amp-story>` and Site Kit can integrate with that to include analytics on the page. In reality, the `amp_analytics_entries` filter could be used for adding the analytics to the page anyway, as is referenced here in a comment (which is not current):

https://github.com/google/site-kit-wp/blob/14d59a5fd00752f47ac955a5b65aacaa31d91b04/includes/Modules/TagManager.php#L160-L163

This may not have been used because `amp_analytics_entries` did not support Tag Manager. For that, we may be dependent on https://github.com/ampproject/amp-wp/pull/2925.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

## Changelog entry

*

**Comment by felixarntz** [comment]
In order to decouple from other work in the AMP plugin, let's rely on the new `amp_print_analytics` action introduced by https://github.com/ampproject/amp-wp/pull/3106. In addition, we should provide backward-compatibility for when that hook doesn't exist yet (e.g. current versions of the AMP plugin).

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by aaemnnosttv** [comment]
Code changes are fairly small to make but it may take a bit longer to QA.

**Comment by tofumatt** [comment]
For clarity: the AMP plugin in question is https://wordpress.org/plugins/amp/

Right now the production version is 1.2.x, so to QA this you'll need to install the production version to test the pre-1.3 codepath.

I went to install 1.3 but the version in the `develop` branch at https://github.com/ampproject/amp-wp is labelled `1.2.3-alpha`. I _assume_ this is the latest version and the one to test with—as there is [no mention of a `1.3` branch](https://github.com/ampproject/amp-wp/branches)

[amp.zip](https://github.com/google/site-kit-wp/files/3576667/amp.zip) is a ZIP file for the development version, built from source. This will be useful if you need to QA the fix for this issue.

**Comment by westonruter** [comment]
That's right. The `develop` branch currently says `1.2.3-alpha` but in reality it is `1.3-alpha`. I just pushed up the version bump onto the `develop` branch to fix this: https://github.com/ampproject/amp-wp/commit/217d5929b15bd48e2eaac431d94b03c198ca5466.

**Comment by adamsilverstein** [comment]
@westonruter Thanks for your work here, this works well in my testing. Moving to "Approval".

### Testing Approach

* Set up site kit with Analytics and Tag Manager, configuring a container.
* Test in each amp mode:
  * on a post and on an amp story
  * with version 1.2.x and version 1.3.x of the amp plugin

In each case review the page source code (using a hard refresh) and verify the tagging appears correctly (once). For paired modes, review the AMP and non amp version.

### Testing Results
Tested with amp plugin from repository, version 1.2.2
* In Reader mode, correct `amp-analytics` tag showed once.
* In Transitional mode, correct `amp-analytics` tag showed once.
* In Standard mode, correct `amp-analytics` tag showed once.

Tested with the `develop` branch (version 1.3-alpha). Created a test story and reviewed in each mode:
* In Reader mode, correct `amp-analytics` tag showed once.
* In Transitional mode, correct `amp-analytics` tag showed once.
* In Standard mode, correct `amp-analytics` tag showed once.

I also re-reviewed the tagging on standard posts under 1.3 and everything looks good.
